frontend/src/features/entity_selector/EntityInput.tsx

```diff
@@ -1,40 +1,163 @@
 import { useCurrentPoll } from 'src/hooks/useCurrentPoll';
 import {
   PRESIDENTIELLE_2022_POLL_NAME,
   YOUTUBE_POLL_NAME,
 } from 'src/utils/constants';
-import { Entity } from 'src/services/openapi';
+import { Entity, UsersService } from 'src/services/openapi';
 import { getAllCandidates } from 'src/utils/polls/presidentielle2022';
+import { getRecommendations } from 'src/features/recommendation/RecommendationApi';
+
+import SelectorListBox, { EntitiesTab } from './EntityTabsBox';
+import SelectorPopper from './SelectorPopper';
+import { videoToEntity } from 'src/utils/video';
+import { useLoginState } from 'src/hooks';
 
 interface Props {
   value: string;
   onChange: (value: string) => void;
+  otherUid: string | null;
 }
 
-const VideoInput = ({ value, onChange }: Props) => {
+const VideoInput = ({ value, onChange, otherUid }: Props) => {
   const { t } = useTranslation();
+  const [suggestionsOpen, setSuggestionsOpen] = useState(false);
+  const inputRef = useRef<HTMLDivElement>(null);
+  const { isLoggedIn } = useLoginState();
+  const fullScreenModal = useMediaQuery(
+    (theme: Theme) => `${theme.breakpoints.down('sm')}, (pointer: coarse)`,
+    { noSsr: true }
+  );
+
+  const handleOptionClick = (uid: string) => {
+    onChange(uid);
+    setSuggestionsOpen(false);
+  };
+
+  const toggleSuggestions = () => setSuggestionsOpen((open) => !open);
+
+  const tabs: EntitiesTab[] = useMemo(
+    () => [
+      {
+        name: 'suggestions',
+        label: t('entitySelector.suggestions'),
+        fetch: async () => {
+          const response = await UsersService.usersMeEntitiesToCompareList({
+            pollName: YOUTUBE_POLL_NAME,
+            limit: 20,
+            firstEntityUid: otherUid || undefined,
+          });
+          return response.results ?? [];
+        },
+        disabled: !isLoggedIn,
+      },
+      {
+        name: 'rate-later',
+        label: t('entitySelector.rateLater'),
+        fetch: async () => {
+          const response = await UsersService.usersMeVideoRateLaterList({
+            limit: 20,
+          });
+          return (response.results ?? []).map((rl) => videoToEntity(rl.video));
+        },
+        disabled: !isLoggedIn,
+      },
+      {
+        name: 'recently-compared',
+        label: t('entitySelector.recentlyCompared'),
+        fetch: async () => {
+          const response = await UsersService.usersMeContributorRatingsList({
+            pollName: YOUTUBE_POLL_NAME,
+            limit: 20,
+          });
+          return (response.results ?? []).map((rating) => rating.entity);
+        },
+        disabled: !isLoggedIn,
+      },
+      {
+        name: 'recommendations',
+        label: t('entitySelector.recommendations'),
+        fetch: async () => {
+          const response = await getRecommendations(
+            'videos',
+            20,
+            '?date=Month',
+            []
+          );
+          return response.results ?? [];
+        },
+      },
+    ],
```

Comment:
I'd really like that we use here the non-connected entities api endpoint for an additional list. If you think it is too much for this PR, let me know and I'll create a ticket for it

Reply:
The loader and error message have been added.

An additional tab is also present to fetch unconnected entities (the tab is disabled if the other entity has not been selected yet).

I noticed that the "unconnected_entities" API could return incorrect results after successive requests. A fix is implemented in #984